3. Условная вероятность
Пусть событие A в примере с мишенью — это попадание в «десятку», событие B —
попадание в любое место мишени. Если известно, что событие B произошло, то ве-
роятность события A повышается. Условная вероятность события A при условии,
что произошло событие B, определяется следующим образом:
 P (AB)
.
P A|B =
P (B)
Если попадание в мишень происходит в восьмидесяти процентах случаев, а в «десят-
ку» — в пяти процентах, то:
P (AB) = P (A) = 0,05
⇒
 0,05
= 0,0625.
P A|B =
0,8

По формуле условной вероятности 0.20/.5=0.4.

In [1]:
.2/.5

0.4

In [2]:
.49+.38+.1+.03

1.0

In [3]:
a = [1,2,3,4]

In [4]:
a*3

[1, 2, 3, 4, 1, 2, 3, 4, 1, 2, 3, 4]

In [5]:
import numpy as np

In [6]:
aa = np.array(a)

In [7]:
aa

array([1, 2, 3, 4])

In [8]:
aa*2

array([2, 4, 6, 8])

создаем вектор вероятностей австралийцев с группами крови
P(I)=0.49,P(II)=0.38,P(III)=0.1,P(IV)=0.03.


Искомая вероятность — сумма вероятностей всех ячеек, отмеченных плюсом. Поскольку австралийцы выбираются независимо, в каждой ячейке вероятность равна произведению вероятностей соответствующих строки и столбца.

In [9]:
a = np.array([0.49, 0.38, 0.1, 0.03])

перемножаем на вероятности по каждой группе получаем вектор

In [44]:
a1 = a * a[0]
a1

array([ 0.2401,  0.1862,  0.049 ,  0.0147])

In [45]:
a2 = a * a[1]
a3 = a * a[2]
a4 = a * a[3]

In [46]:
print a1
print a2
print a3
print a4

[ 0.2401  0.1862  0.049   0.0147]
[ 0.1862  0.1444  0.038   0.0114]
[ 0.049  0.038  0.01   0.003]
[ 0.0147  0.0114  0.003   0.0009]


In [47]:
[b for b in a]

[0.48999999999999999, 0.38, 0.10000000000000001, 0.029999999999999999]

проверяем справедливось условия полной группы

In [48]:
sum = 0.

In [49]:
for b in a1: sum = sum + b
for b in a2: sum = sum + b
for b in a3: sum = sum + b
for b in a4: sum = sum + b
sum

1.0

выбираем только совместимые вероятности и суммируем как независимые

In [50]:
result = 0.
for b in a1: result = result + b
result = result + a2[1] + a2[3]
result = result + a3[2] + a3[3]
result = result + a4[3]
result


0.65969999999999995

ответ P = 0.65969999999999995 округлить? 0.6597

В супермаркете 60% яблок из Турции и 40% яблок из Индии. 10% турецких и 15% индийских яблок — червивые. Какова вероятность, что яблоко, купленное в этом магазине, окажется червивым? Запишите ответ от 0 до 1 с двумя знаками после десятичной точки.

событие выбор одного яблока независимое pt = 0.6 pi = 0.4
общее кол-во хороших яблок независимое для обоих типов
90% турецких
85% индийских
умножение вероятностей дает вероятность хоорших яблок

Пусть событие A происходит, если яблоко из Турции, а событие B — если яблоко червивое. Воспользуемся формулой полной вероятности:

𝖯(B)=𝖯(B∣∣A)𝖯(A)+𝖯(B∣∣A¯)𝖯(A¯)=0.1⋅0.6+0.15⋅0.4=0.12.

In [34]:
pt = .9 * .6
pi = .85 * .4

суммируем получаем вероятность хорошего яблока

In [35]:
pgood = pt + pi

вероятность плохого по формуле полной группы

In [37]:
pbad = 1 - pgood
print pbad

0.12


Пусть событие A происходит, если яблоко из Турции, а событие B — если яблоко червивое. Воспользуемся формулой полной вероятности:

𝖯(B)=𝖯(B∣∣A)𝖯(A)+𝖯(B∣∣A¯)𝖯(A¯)=0.1⋅0.6+0.15⋅0.4=0.12.

Представьте, что вы купили яблоко в магазине из предыдущей задачи, и оно оказалось червивым. Какова вероятность того, что оно из Турции, а не из Индии?

событие червивое яблоко, вероятность из какой группы из условия 60% турецких нет не верно нужно использовать формулу условной вероятности
P(A|B) = P(AB)/P(B) = 

In [57]:
pt = (.1*.15)/.15
pt

0.1

Журнал "Популярная механика" читает 6.4% целевой аудитории вашего продукта, а журнал "Кот Шрёдингера" — 3.7%. Если вы разместите рекламу в обоих журналах, что можно сказать о доле p целевой аудитории, которая увидит рекламу хотя бы один раз?

Используйте формулу вероятности события-суммы и подумайте, чему может быть равна вероятность события-произведения.


событие суммы P(A) + P(B)

In [52]:
.064+.037

0.101

событие произведения P(AB) = 0.037 фиг с маслом неправильно
увидит хотябы один раз P(AB) = 0.064 в этом случае?

In [58]:
ptc = .1 * .6
pic = .15 * .4

In [59]:
ptc+pic

0.12

In [61]:
ptc/(ptc+pic)

0.5

In [62]:
ptc

0.06

In [63]:
pic

0.06

1% женщин больны раком груди. У 80% женщин, больных раком груди, маммограмма верно выявляет наличие заболевания; кроме того, она даёт ложный положительный результат (то есть, неверно показывает наличие рака) для 9.6% здоровых женщин.

У какого процента женщин, маммограмма которых дала положительный результат, есть рак груди? Запишите ответ с точностью до одного знака после десятичной точки (знак процента не нужен).

дано: событие - тест для любой из 0.99+0.01, результат положительный
условие: положительный верный для 0.8 больных и положительный не верный для 0.096 здоровых
найти: количество с результатом положительный и верный для какого количества женщин  в %
решение: 


In [2]:
0.99*0.096 # количество здоровых с неверным результатом

0.09504

In [3]:
0.01*0.8 # количество больных с верным езультатом 

0.008

In [6]:
0.09504+0.008 # общее количество  положительным результатом

0.10303999999999999

In [7]:
0.008/(0.09504+0.008)  # количество в процентах реально больных

0.07763975155279504

In [ ]:
7.8

Какова вероятность того, что при независимом подбрасывании двух симметричных шестигранных кубиков хотя бы на одном из них выпадет больше трёх очков? Запишите точный ответ в виде десятичной дроби.

4 1/6, 5 1/6, 6 1/6 итого 3/6 или 1/2 для одного кубика 
P(A+B)=P(A)+P(B)-P(AB) или 3/4

In [9]:
3./4.

0.75